chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -1281,6 +1317,21 @@ impl actix::Handler<WithSpanContext<Tier3Request>> for PeerManagerActor {
         ctx.spawn(wrap_future(
             async move {
                 let tier3_response = match request.body {
+                    Tier3RequestBody::StateHeader(StateHeaderRequestBody { shard_id, sync_hash }) => {
+                        match state.client.send_async(StateRequestHeader { shard_id, sync_hash }).await {
+                            Ok(Some(client_response)) => {
+                                PeerMessage::VersionedStateResponse(*client_response.0)
+                            }
+                            Ok(None) => {
+                                tracing::debug!(target: "network", "client declined to respond to {:?}", request);
+                                return;
+                            }
+                            Err(err) => {
+                                tracing::error!(target: "network", ?err, "client failed to respond to {:?}", request);
+                                return;
+                            }
```

Comment:
For my information why may a peer reject a request? What may cause an error? In those cases do we still want to notify the other node that their request was rejected / failed?

Reply:
We implement some [throttling on the client side](https://github.com/near/nearcore/blob/f8b6520a73a4ac13687ac2944b8e4883c7d387ef/chain/client/src/view_client_actor.rs#L1341-L1344) to protect the local resources of the node if too many requests arrive over the network at once. I'll update the log message to clarify this.

We don't bother to notify the other node because requests can fail silently for other reasons such a network message loss. The requesting node implements a time-out and will retry from other hosts as needed.

Reply:
An error response could be sent over Tier2; network resource consumption is not a prohibitive factor here.

The main thing is we don't have any reason to behave differently on the requesting side whether the request comes back with an error or if it's lost in the network.

Even re-requesting faster in the error case is not necessarily desirable since state sync is not strictly optimizing for time-to-completion and rather we want to spread out load on the network.